apps/src/aiDifferentiation/AiDiffFloatingActionButton.tsx

```diff
@@ -54,6 +55,23 @@ const AiDiffFloatingActionButton: React.FC<AiDiffFloatingActionButtonProps> = ({
       isFirstSession
   );
 
+  const [curriculumCourses, setCurriculumCourses] = useState();
+
+  useEffect(() => {
+    const body = JSON.stringify({
+      context: context,
+      contextId: scriptId,
+    });
+    HttpClient.post(`/ai_diff/curriculum_courses`, body, true, {
```

Comment:
Do we ever have this loaded into redux? i.e. the teacher dashboard generally has unit/course data already loaded in. Is there a way that we can use what is already loaded into redux by default and only use this endpoint when we don't have the info on the page?